base/array.jl

```diff
@@ -711,8 +711,8 @@ function find(testf::Function, A::AbstractArray)
     # use a dynamic-length array to store the indexes, then copy to a non-padded
     # array for the return
     tmpI = Array(Int, 0)
-    for i = 1:length(A)
-        if testf(A[i])
+    for (i,j) = enumerate(eachindex(A))
+        if testf(A[j])
```

Comment:
I see, one doesn't need the index of A at all, I will update. But just out of curiosity: is this a bug? The code worked for me, are there cases where it doesn't?